Classes/common/HTTP/CDTURLSessionInterceptorContext.h

```diff
@@ -0,0 +1,39 @@
+//
+//  CDTURLSEssionFilterContext.h
+//  
+//
+//  Created by Rhys Short on 17/08/2015.
+//
+//
+
+#import <Foundation/Foundation.h>
+#import "CDTMacros.h"
+
+@interface CDTURLSessionInterceptorContext : NSObject
+
+@property (nonnull, readonly, nonatomic, strong) NSMutableURLRequest *request;
```

Comment:
I get build errors on the `nonnull` annotation. This probably means I'm not on a bleeding edge XCode but I'm sure we still need to support older versions - is this something we can fix with a compatibility macro or similar?